### PR TITLE
feat(pie-modal): DSW-931 added supportingAction prop

### PR DESCRIPTION
**Pull request by kevinrodrigues**
## Describe your changes (can list changeset entries if preferable)

[Added] SupportingAction prop to modal

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-modal/src/index.ts [hunk 1]
```diff
@@ -233,7 +240,7 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
      * @private
      */
     private renderLeadingAction (): TemplateResult | typeof nothing {
```
Comment (raoufswe) [inline packages/components/pie-modal/src/index.ts:242]: i might be missing something here, but don't we want to allow users to trigger an on click callballback for the action buttons here? they both close the modal at the moment.

File: packages/components/pie-modal/src/index.ts [hunk 2]
```diff
@@ -251,6 +258,41 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
...
+
+        if (!this.leadingAction) {
+            console.warn('Use `leadingAction` instead of `supportingAction`. `supportingAction` is being ignored.');
+            return nothing;
+        }
+
+        return html`
+              <pie-button
+                variant="${variant}"
+                aria-label="${ariaLabel || nothing}"
```
Comment (raoufswe) [inline packages/components/pie-modal/src/index.ts:287]: can we use ifDefined here?
https://lit.dev/docs/templates/directives/#ifdefined

Comment (kevinrodrigues) [reply packages/components/pie-modal/src/index.ts:287]: I tried a few things before i went this route initially. The issue is with `ifDefined` and `guard` they still render the attribute itself if the value is empty i.e `ariaLabel: ''`. With `nothing` if the ariaLabel is empty or doesn't exist then the attribute won't render.

File: packages/components/pie-modal/src/index.ts (hunk 1 above)
Comment (kevinrodrigues) [reply packages/components/pie-modal/src/index.ts:242]: Not sure if there is a ticket for this but the markup that was there before both closed the dialog, I'm guessing this is just intentional until we have a tickets for this work. @xander-marjoram @fernandofranca do you know if we have one? If not i can add the changes here.

**Review by fernandofranca — APPROVED** [review]

**Pull request merged**